mr/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java

```diff
@@ -202,12 +208,31 @@ public static PartitionSpec spec(Configuration config) {
    */
   @VisibleForTesting
   static void put(Configuration config, Table table) {
+    // The Table contains a FileIO and the FileIO serializes the configuration so we might end up recursively
+    // serializing the objects. To avoid this unset the values for now before serializing.
+    config.unset(InputFormatConfig.SERIALIZED_TABLE);
+    config.unset(InputFormatConfig.FILE_IO);
+    config.unset(InputFormatConfig.LOCATION_PROVIDER);
+    config.unset(InputFormatConfig.ENCRYPTION_MANAGER);
+    config.unset(InputFormatConfig.TABLE_LOCATION);
+    config.unset(InputFormatConfig.TABLE_SCHEMA);
+    config.unset(InputFormatConfig.PARTITION_SPEC);
+
+    String base64Table = table instanceof Serializable ? SerializationUtil.serializeToBase64(table) : null;
+    String base64Io = SerializationUtil.serializeToBase64(table.io());
+    String base64LocationProvider = SerializationUtil.serializeToBase64(table.locationProvider());
+    String base64EncryptionManager = SerializationUtil.serializeToBase64(table.encryption());
+
+    if (base64Table != null) {
+      config.set(InputFormatConfig.SERIALIZED_TABLE, base64Table);
+    }
+
+    config.set(InputFormatConfig.FILE_IO, base64Io);
+    config.set(InputFormatConfig.LOCATION_PROVIDER, base64LocationProvider);
+    config.set(InputFormatConfig.ENCRYPTION_MANAGER, base64EncryptionManager);
```

Comment:
Shouldn't these be serialized only if the table is not?

Reply:
Let's see what we can do about the serialization of the other table types. I would love to fully get rid of these serialized objects. Maybe even on the expense of not supporting reading some types of the tables through mr.